### ISSUE TITLE
 Avoid UnusedPrivateMember in `serialize only data class`

### ISSUE THREAD

**Issue by zmunm** [detekt/detekt#3165]
## Expected Behavior of the rule

Today we have a very interesting situation. At least I think so.
I hope someone has used retrofit. I wanted to give a more general example, but I didn't come up with a suitable one. I'm sorry.

UnusedPrivateMember is not required in a specific class that is used only when serializing. So I want to filter it.

## Context

retofit api interface
```kotlin
    @POST("api/addresses/")
    fun post(@Body body: RequestAddress): Single<ResponseAddress>
```

RequestAddress.
```kotlin
data class RequestAddress(
    private val name: String,
    private val address: String,
    private val mobile: String
)
```
There is no need to make the field public in this class. JsonParser such as Gson serializes RequestAddress class, but in this case, I do not need to directly access the field, so I want to keep the field private.

This triggers UnusedPrivateMember. How can I avoid it?

**Comment by schalkms** [comment]
Hey! Thanks for the feedback.

For me this seems like a framework specific issue.
Detekt decided to not include framework specifics into the implementation of rules.

> UnusedPrivateMember is not required in a specific class that is used only when serializing. So I want to filter it.

How would you want to detect this? Search all classes that are referenced in a REST API like function would maybe be a solution.
Anyway, even if there's a simple solution, I think value/effort is not given here.

> This triggers UnusedPrivateMember. How can I avoid it?

1. The configuration option `allowedNames` allows to ignore unused members. Maybe it's an option to ignore `body` here.
2. An `Ignore` annotation ca be used.
3. If you want to, you can also extend this rule in a [custom domain-specific rule](https://detekt.github.io/detekt/extensions.html#custom-rulesets).

Due to the mentioned arguments and solutions, I'll close this issue.

**Comment by zmunm** [comment]
Thank you!

**Comment by schalkms** [comment]
You are welcome! 🙂
